### PR TITLE
enable mcm feature during upgrade

### PR DESCRIPTION
**Pull request by ibrokethecloud**
**IMPORTANT: Please do not create a Pull Request without creating an issue first.**

**Problem:**

PR: https://github.com/harvester/harvester-installer/pull/468 introduced changes for enabling rancher mcm as part of new install.

**Solution:**

The change introduces a minor change to the upgrade-manifests.sh which sets `multi-cluster-management=true` during rancher upgrade.

**Related Issue:**
https://github.com/harvester/harvester/issues/2679
**Test plan:**

### CODE REVIEW

File: package/upgrade/upgrade_manifests.sh [hunk 1]
```diff
@@ -229,6 +229,8 @@ upgrade_rancher() {
   cd $UPGRADE_TMP_DIR/rancher
 
   ./helm get values rancher -n cattle-system -o yaml >values.yaml
+  echo "Patching rancher features"
+  yq e -i '.features= "multi-cluster-management=true,multi-cluster-management-agent=false"' values.yaml
```
Comment (starbops) [inline package/upgrade/upgrade_manifests.sh]: Could we move these two lines below the `cat values.yaml`. I believe the original intention of presenting values here is to ensure what the current situation is before upgrading Rancher, then we do the patches.

**Review by starbops — COMMENTED** [review]
LGTM, just one nit. Thanks!

**Comment by bk201** [comment]
On a 3-node cluster, I'm always seeing the first apply-manifest pod fail:
Screenshot 2023-05-17 at 11 00 14 PM

First failing job pod's log:
```
2023-05-17T14:54:19.199152298Z stdout F systemDefaultRegistry: ""
2023-05-17T14:54:19.199156065Z stdout F tls: external
2023-05-17T14:54:19.199159932Z stdout F useBundledSystemChart: true
2023-05-17T14:54:19.827558418Z stdout F CAPI cluster fleet-local/local is provisioned (current generation: 376).
2023-05-17T14:54:21.117199662Z stdout F clusterrepo.catalog.cattle.io "rancher-charts" deleted
2023-05-17T14:54:21.553376355Z stdout F clusterrepo.catalog.cattle.io "rancher-rke2-charts" deleted
2023-05-17T14:54:21.985028984Z stdout F clusterrepo.catalog.cattle.io "rancher-partner-charts" deleted
2023-05-17T14:54:22.385808157Z stdout F setting.management.cattle.io "chart-default-branch" deleted
2023-05-17T14:59:25.707957503Z stderr F Error: UPGRADE FAILED: timed out waiting for the condition
```



This is due to the old rancher pod taking too long to switch to the new version (helm upgrade's default timeout is 5 minutes). Seeing this in its log:
```
2023/05/17 14:58:14 [FATAL] feature flag [multi-cluster-management] value has changed, rancher must be restarted  
```


Increasing the upgrade timeout seems to solve this issue:

```
./helm upgrade rancher ./*.tgz --namespace cattle-system -f values.yaml --timeout 10m --wait 
```
([code](https://github.com/harvester/harvester/blob/25d478c02387de0fbafc307c0a8fcb4e16ee2d48/package/upgrade/upgrade_manifests.sh#L259))

**Review by bk201 — APPROVED** [review]
lgtm

**Review by starbops — APPROVED** [review]
Tested 2-node upgrade, looking good.

**Review by guangbochen — APPROVED** [review]

**Pull request merged**